File: app/Models/Lokasi.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Lokasi extends Model
{
    use HasFactory;
    protected $table = 'lokasi';
    protected $primaryKey = 'id_lokasi';

    protected $fillable = [
        'id_mitra',
        'nama_lokasi',
        'alamat',
        'path_area',
        'deskripsi'

    ];

    public function mitra()
    {
        return $this->belongsTo(Mitra::class, 'id_mitra');
    }
    public function area_gudang()
    {
        return $this->hasMany(AreaGudang::class, 'id_lokasi');
    }
    public function rating_mitra()
    {
        return $this->hasMany(RatingMitra::class, 'id_lokasi');
    }
}

File: lib/model/lokasi.dart
import 'mitra.dart';
import 'areagudang.dart';
class Lokasi {
  final int idLokasi;
  final String namaLokasi;
  final String deskripsi;
  final Mitra? mitra;
  final List<AreaGudang> areaGudang;

  Lokasi({
    required this.idLokasi,
    required this.namaLokasi,
    required this.deskripsi,
    this.mitra,
    required this.areaGudang,
  });

  factory Lokasi.fromJson(Map<String, dynamic> json) {
    return Lokasi(
      idLokasi: json['id_lokasi'],
      namaLokasi: json['nama_lokasi'] ?? "",
      deskripsi: json['deskripsi'] ?? "",
      mitra: json['mitra'] != null ? Mitra.fromJson(json['mitra']) : null,
      areaGudang: (json['area_gudang'] as List<dynamic>? ?? [])
          .map((e) => AreaGudang.fromJson(e))
          .toList(),
    );
  }
}
